subscription documentation (#164)
valkey-io/valkey-swift: Sources/Valkey/Documentation.docc/Pubsub.md

```swift
try await connection.subscribe(to: ["channel*"]) { subscription in
    for try await entry in subscription {
        let channel = "\(entry.channel)/last"
        try await connection.set(channel, value: entry.message)
    }
}
```

The code above receives all messages sent to channels prefixed with the string "channel".

More can be found out about Valkey pub/sub in the [Valkey documentation](https://valkey.io/topics/pubsub/).

### Support for Client Side Caching

Client side caching is a way to improve performance of a service using Valkey. Caching the values of specific keys locally avoids putting pressure on your Valkey server unnecessarily. For a client side cache to work though, we need to know when to invalidate our local data. Valkey provides two different ways to do this. Both of these work using pub/sub.

1) The server remembers the keys a connection has accessed and publishes events to the invalidation channel whenever those keys are modified.

> 2) Broadcasting, where the server doesn't keep a record of what keys have been accessed, instead the client provides a prefix of the keys they are interested in. The invalidation channel will then receive events whenever any key with that prefix is modified.

Reviewer: ```suggestion
2) Broadcasting, where the server doesn't keep a record of what keys have been accessed. The client provides a prefix of the keys they are interested in and the server sends events to the invalidation channel whenever any key that matches that prefix is modified.
```

#### Enabling Tracking

> Connections start without invalidation tracking enabled. To enable receiving invalidation events on a connection. You need to call

Reviewer: ```suggestion
Connections start without invalidation tracking enabled. To enable receiving invalidation events on a connection, call ``ValkeyClientProtocol/clientTracking(status:clientId:prefixs:bcast:optin:optout:noloop:)``. For example:
```

```swift
try await connection.clientTracking(status: .on)
```

> This will enable method 1 where the server remembers the keys accessed by a connection. If you would like to track changes to all keys with a prefix, you can use

Reviewer: ```suggestion
This tells the server to use the first invalidation method where it remembers the keys accessed by a connection. If you would like to track changes to all keys with a prefix, use:
```

```swift
try await connection.clientTracking(
    status: .on,
    prefixes: ["object:"],
    bcast: true
)
```

#### Subscribing to Invalidation Events

Once tracking is enabled you can subscribe to invalidation events using ``ValkeyConnection/subscribeKeyInvalidations(process:)``. The AsyncSequence passed to the `process` closure is a list of keys that have been invalidated.

```swift
try await connection.subscribeKeyInvalidations { keys in
    for try await key in keys {
        myCache.invalidate(key)
    }
}
```

#### Redirecting Invalidation Events

> With RESP3 it is possible to perform data operations and receive the invalidation events on the same connection, but Valkey client tracking also allows you to redirect invalidation events to another connection. Given valkey-swift uses a persistent connection pool it is probably preferable to use a single connection for invalidation message if you want to implement a system wide cache.

Reviewer: ```suggestion
With RESP3 it is possible to perform data operations and receive the invalidation events on the same connection, but Valkey client tracking also allows you to redirect invalidation events to another connection. Given that the client uses a persistent connection pool, it is preferable to use a single connection for invalidation message to implement a system wide cache.
```

> For this to work you need to know the id of the connection that is subscribed to the key invalidation events. You can get connection id using ``ValkeyClientProtocol/clientId()`` and when setting up tracking you include the client id. 

Reviewer: ```suggestion
For this to work you need to know the id of the connection that is subscribed to the key invalidation events. Get the connection id using ``ValkeyClientProtocol/clientId()`` and use it when you set up tracking. 
```
